tests/integration/test_bayesian_optimization.py

```diff
@@ -354,104 +352,81 @@ def _test_optimizer_finds_minimum(
         minimizers = SIMPLE_QUADRATIC_MINIMIZER
         minima = SIMPLE_QUADRATIC_MINIMUM
         rtol_level = 0.05
+    if model_type in ["SVGP", "DGP", "DE"]:
+        num_initial_query_points = 20
 
-    if model_type in ["GPR", "VGP", "SVGP"]:
-
-        def build_model(data: Dataset) -> GPflowPredictor:  # type: ignore
-            assert model_args is not None
+    initial_query_points = search_space.sample(num_initial_query_points)
+    observer = mk_observer(scaled_branin if optimize_branin else simple_quadratic)
+    initial_data = observer(initial_query_points)
 
-            variance = tf.math.reduce_variance(data.observations)
-            kernel = gpflow.kernels.Matern52(variance, tf.constant([0.2, 0.2], tf.float64))
-            scale = tf.constant(1.0, dtype=tf.float64)
-            kernel.variance.prior = tfp.distributions.LogNormal(
-                tf.constant(-2.0, dtype=tf.float64), scale
-            )
-            kernel.lengthscales.prior = tfp.distributions.LogNormal(
-                tf.math.log(kernel.lengthscales), scale
-            )
-
-            if model_type == "GPR":
-                gpr = gpflow.models.GPR(
-                    (data.query_points, data.observations), kernel, noise_variance=1e-5
-                )
-                gpflow.utilities.set_trainable(gpr.likelihood, False)
-                return GaussianProcessRegression(gpr, **model_args)
-            elif model_type == "VGP":
-                likelihood = gpflow.likelihoods.Gaussian(1e-3)
-                vgp = gpflow.models.VGP(initial_data.astuple(), kernel, likelihood)
-                gpflow.utilities.set_trainable(vgp.likelihood, False)
-                return VariationalGaussianProcess(vgp, **model_args)
-            elif model_type == "SVGP":
-                Z = search_space.sample_sobol(50)  # Initialize diverse inducing locations
-                svgp = gpflow.models.SVGP(
-                    kernel,
-                    gpflow.likelihoods.Gaussian(variance=1e-5),
-                    Z,
-                    num_data=len(data.observations),
-                )
-                gpflow.utilities.set_trainable(svgp.likelihood, False)
-                return SparseVariational(svgp, **model_args)
+    if model_type == "GPR":
+        if "LocalPenalization" in acquisition_rule.__repr__():
+            likelihood_variance = 1e-3
+        else:
+            likelihood_variance = 1e-5
+        gpr = build_gpr(initial_data, search_space, likelihood_variance=likelihood_variance)
+        model = GaussianProcessRegression(gpr, **model_args)
```

Comment:
If you mark this as a GPflowPredictor then you shouldn't need the type ignores below (except for the Keras model) plus it corresponds to the model type we claim in GPR_OPTIMIZER_PARAMS

```suggestion
        model: GPflowPredictor = GaussianProcessRegression(gpr, **model_args)
```

That said the test is currently a bit of a typing mess, as there are already some rules that only work with GPR rather than all GPflowPredictors. And adding Keras made it worse, as many more of the rules don't work with that. I might refactor it in a separate PR.

Reply:
yeah, its a bit of a mess in terms of typing, but also doesnt make sense to duplicate whole BO loop part between the models...

Reply:
that suggestion raises some other typing issues (in optimize on line 425...), so perhaps we keep it as is for now?